src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php

```diff
@@ -144,8 +145,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
             ? new TargetOperation($currentCatalogue, $extractedCatalogue)
             : new MergeOperation($currentCatalogue, $extractedCatalogue);
 
+        $domains = $operation->getDomains();
+        $domain = $input->getOption('domain');
+
+        if (null !== $domain) {
+            $domains = array_intersect($domains, array($domain));
```

Comment:
I'll leave the ``array_intersect()`` because if you use for example ``--domain nonexisting``
Then you'll get an empty array when calling ``getDomains()`` so the command will exit (L155-158)